File: cdk-from-cfn-testing/expected/ec2_encryption/golang/Ec2EncryptionConstruct.go
package main

import (
	cdk "github.com/aws/aws-cdk-go/awscdk/v2"
	ec2 "github.com/aws/aws-cdk-go/awscdk/v2/awsec2"
	"github.com/aws/constructs-go/constructs/v10"
	"github.com/aws/jsii-runtime-go"
)

type Ec2EncryptionConstructProps struct {
	Environment *string
	DatabaseType *string
	UseEncryption interface{/* Boolean */}
	EncryptedAmi *string
	UnencryptedAmi *string
	SubnetType *string
	EnableMonitoringParameter interface{/* Boolean */}
}

type Ec2EncryptionConstruct struct {
	constructs.Construct
}

func NewEc2EncryptionConstruct(scope constructs.Construct, id string, props *Ec2EncryptionConstructProps) *Ec2EncryptionConstruct {
	regionToAmi := map[*string]map[*string]*string{
		jsii.String("us-east-1"): map[*string]*string{
			jsii.String("AMI"): jsii.String("ami-0c02fb55956c7d316"),
		},
		jsii.String("us-west-2"): map[*string]*string{
			jsii.String("AMI"): jsii.String("ami-008fe2fc65df48dac"),
		},
		jsii.String("eu-west-1"): map[*string]*string{
			jsii.String("AMI"): jsii.String("ami-0c9c942bd7bf113a2"),
		},
		jsii.String("ap-southeast-1"): map[*string]*string{
			jsii.String("AMI"): jsii.String("ami-0c802847a7dd848c0"),
		},
		jsii.String("us-east-2"): map[*string]*string{
			jsii.String("AMI"): jsii.String("ami-0900fe555666598a2"),
		},
	}

	construct := constructs.NewConstruct(scope, &id)

	hasDatabase := props.DatabaseType == jsii.String("mysql")

	isProduction := props.Environment == jsii.String("prod")

	usePrivateSecurityGroup := props.SubnetType == jsii.String("Private1") || props.SubnetType == jsii.String("Private2")

	keyPairProd := !isProduction

	useEncryption := isProduction && hasDatabase

	privateSecurityGroup := ec2.NewCfnSecurityGroup(
		construct,
		jsii.String("PrivateSecurityGroup"),
		&ec2.CfnSecurityGroupProps{
			GroupDescription: jsii.String("Private security group"),
		},
	)

	publicSecurityGroup := ec2.NewCfnSecurityGroup(
		construct,
		jsii.String("PublicSecurityGroup"),
		&ec2.CfnSecurityGroupProps{
			GroupDescription: jsii.String("Public security group"),
		},
	)

	ec2.NewCfnInstance(
		construct,
		jsii.String("MyApp"),
		&ec2.CfnInstanceProps{
			ImageId: regionToAmi[cdk.Stack_Of(construct).Region()][jsii.String("AMI")],
			InstanceType: jsii.String("t3.micro"),
			Tags: &[]*cdk.CfnTag{
				&cdk.CfnTag{
					Key: jsii.String("Name"),
					Value: cdk.Fn_Select(jsii.Number(1), cdk.Fn_Split(jsii.String("-"), jsii.String("My-EC2-Instance"))),
				},
			},
			SecurityGroups: &[]*string{
				ifCondition(
					usePrivateSecurityGroup,
					privateSecurityGroup.Ref(),
					publicSecurityGroup.Ref(),
				),
			},
		},
	)

	return &Ec2EncryptionConstruct{
		Construct: construct,
	}
}

/// ifCondition is a helper function that replicates the ternary
/// operator that can be found in other languages. It is conceptually
/// equivalent to writing `cond ? whenTrue : whenFalse`, meaning it
/// returns `whenTrue` if `cond` is `true`, and `whenFalse` otherwise.
func ifCondition[T any](cond bool, whenTrue T, whenFalse T) T {
	if cond {
		return whenTrue
	}
	return whenFalse
}

File: cdk-from-cfn-testing/expected/ec2_encryption/python/Ec2EncryptionConstruct.py
from aws_cdk import Stack
import aws_cdk as cdk
import aws_cdk.aws_ec2 as ec2
from constructs import Construct

class Ec2EncryptionConstruct(Construct):
  def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
    super().__init__(scope, construct_id)

    # Applying default props
    props = {
      'environment': kwargs.get('environment', 'dev'),
      'databaseType': kwargs.get('databaseType', 'postgresql'),
      'useEncryption': kwargs.get('useEncryption', False),
      'encryptedAmi': kwargs.get('encryptedAmi', 'ami-1234567890abcdef0'),
      'unencryptedAmi': kwargs.get('unencryptedAmi', 'ami-0987654321fedcba0'),
      'subnetType': kwargs.get('subnetType', 'Private1'),
      'enableMonitoringParameter': kwargs.get('enableMonitoringParameter', False),
    }

    # Mappings
    regionToAmi = {
      'us-east-1': {
        'AMI': 'ami-0c02fb55956c7d316',
      },
      'us-west-2': {
        'AMI': 'ami-008fe2fc65df48dac',
      },
      'eu-west-1': {
        'AMI': 'ami-0c9c942bd7bf113a2',
      },
      'ap-southeast-1': {
        'AMI': 'ami-0c802847a7dd848c0',
      },
      'us-east-2': {
        'AMI': 'ami-0900fe555666598a2',
      },
    }

    # Conditions
    has_database = props['databaseType'] == 'mysql'
    is_production = props['environment'] == 'prod'
    use_private_security_group = (props['subnetType'] == 'Private1' or props['subnetType'] == 'Private2')
    key_pair_prod = not is_production
    use_encryption = (is_production and has_database)

    # Resources
    privateSecurityGroup = ec2.CfnSecurityGroup(self, 'PrivateSecurityGroup',
          group_description = 'Private security group',
        )

    publicSecurityGroup = ec2.CfnSecurityGroup(self, 'PublicSecurityGroup',
          group_description = 'Public security group',
        )

    myApp = ec2.CfnInstance(self, 'MyApp',
          image_id = regionToAmi[Stack.of(self).region]['AMI'],
          instance_type = 't3.micro',
          tags = [
            {
              'key': 'Name',
              'value': cdk.Fn.select(1, 'My-EC2-Instance'.split('-')),
            },
          ],
          security_groups = [
            privateSecurityGroup.ref if use_private_security_group else publicSecurityGroup.ref,
          ],
        )


